### PR TITLE
cli: implement start command

### LINKED ISSUE #95: cli: implement `start` functionality
```console
$ reana-client start --help
Usage: reana-client start [OPTIONS]

  Start previously created workflow.

  The ``start`` command allows to start previously created workflow. The
  workflow execution can be further influenced by passing input prameters
  using ``-p`` or ``--parameters`` flag and by setting additional operational
  options using ``-o`` or ``--options``.  The input parameters and operational
  options can be repetitive. For example, to disable caching for the Serial
  workflow engine, you can set ``-o CACHE=off``.

  Examples:

       $ reana-client start -w myanalysis.42 -p sleeptime=10 -p myparam=4

       $ reana-client start -w myanalysis.42 -p myparam1=myvalue1 -o CACHE=off

Options:
  -w, --workflow TEXT      Name or UUID of the workflow. Overrides value of
                           REANA_WORKON environment variable.
  -t, --access-token TEXT  Access token of the current user.
  -p, --parameter TEXT     Additional input parameters to override original
                           ones from reana.yaml. E.g. -p myparam1=myval1 -p
                           myparam2=myval2.
  -o, --option TEXT        Additional operational options for the workflow
                           execution. E.g. CACHE=off. (workflow engine -
                           serial) E.g. --debug (workflow engine - cwl)
  --follow                 If set, follows the execution of the workflow until
                           termination.
  --help                   Show this message and exit.
```

Here the complexity will be the multiple handling of parameters and options, and the waiting for output when the `--follow` option is used. Some looping with a safety timeout will be necessary there.

Example scenario:

```console
$ cd reana-demo-root6-roofit
$ reana-client create -w demo
$ reana-client upload -w demo
$ reana-client start -w demo --follow
==> SUCCESS: demo.4 is pending
==> SUCCESS: demo.4 is pending
==> SUCCESS: demo.4 is pending
==> SUCCESS: demo.4 is running
==> SUCCESS: demo.4 is running
==> SUCCESS: demo.4 is running
==> SUCCESS: demo.4 has been finished
==> Listing workflow output files...
https://localhost:30443/api/workflows/demo.4/workspace/code/gendata.C
https://localhost:30443/api/workflows/demo.4/workspace/code/fitdata.C
https://localhost:30443/api/workflows/demo.4/workspace/results/data.root
https://localhost:30443/api/workflows/demo.4/workspace/results/plot.png
```

For testing, the first two `create` and `upload` commands should be used from the Python client, and then `start` from the new Go client.

### PR DESCRIPTION
**Pull request by BrunoRosendo**
closes #95

### CODE REVIEW

File: cmd/start.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+) error {
+	for slices.Contains([]string{"pending", "queued", "running"}, currentStatus) {
+		time.Sleep(time.Duration(config.CheckInterval) * time.Second)
+		status, err := workflows.GetStatus(token, workflow)
+		if err != nil {
+			return err
+		}
+		currentStatus = status.Status
+
+		statusMsg, err := workflows.StatusChangeMessage(workflow, currentStatus)
```
Comment (mdonadoni) [inline cmd/start.go:204]: I think some of the messages are not correct, like `workflow has been failed` or `workflow has been finished`. What do you think about opening an issue, for both this client and the python one, to improve these messages?

File: cmd/start.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+		if err != nil {
+			return err
+		}
+		displayer.DisplayMessage(
+			statusMsg,
+			displayer.Success,
+			false,
+			cmd.OutOrStdout(),
+		)
```
Comment (mdonadoni) [inline cmd/start.go:213]: When the workflow fails, the client incorrectly shows `==> SUCCESS: workflow has been failed`. The python client has the same behaviour. What about opening an issue to fix this?

File: cmd/start.go [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+				workflow:    workflow,
+				displayURLs: true,
+				page:        1,
+			}
+			err = lsParams.run(cmd)
+			if err != nil {
+				return err
+			}
+		} else if slices.Contains([]string{"deleted", "failed", "stopped"}, currentStatus) {
+			return errors.New("the workflow stopped abruptly")
```
Comment (mdonadoni) [inline cmd/start.go:234]:
```suggestion
			return errors.New("the workflow did not finish")
```
WDYT?

File: cmd/start.go (hunk 1 above)
Comment (BrunoRosendo) [reply cmd/start.go:204]: I agree, I also thought the messaged didn't work very well but recreated the Python behavior anyway. I'll create issues for that

File: cmd/start.go (hunk 2 above)
Comment (BrunoRosendo) [reply cmd/start.go:213]: I agree, I'll open the issues

**Review by mdonadoni — APPROVED** [review]
LGTM!

**Pull request merged**